docs/development/mkdocs.md

```diff
@@ -0,0 +1,67 @@
+# Creating a Documentation for DevStream
+
+## Background
+
+1. We use [readthedocs](https://readthedocs.org/) to host our documentation. You do not need to know too much more about it in order to create a doc for DevStream, though.
+1. Readthedocs supports both [`sphinx`](https://docs.readthedocs.io/en/stable/intro/getting-started-with-sphinx.html) and [`mkdocs`](https://docs.readthedocs.io/en/stable/intro/getting-started-with-mkdocs.html) to build the doc site; we use `mkdocs` at the moment. If you meet any issues, the official doc of `mkdocs` [here](https://www.mkdocs.org/) is a good place to start.
+1. In our mkdocs setup, we use the `material` theme. More info on "material":
+    - [website](https://squidfunk.github.io/mkdocs-material/)
+    - [docs](https://squidfunk.github.io/mkdocs-material/getting-started/)
+    - [GitHub repo](https://github.com/squidfunk/mkdocs-material)
+1. We also use two plugins for mkdocs:
+    - [search](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/)
+    - [mkdocs-i18n](https://pypi.org/project/mkdocs-i18n/)
+
+## Prerequisites
+
+- Python3 (`mkdocs` is a Python thing)
+- pip3 (to install dependencies)
+
+Suggested setup for macOS users:
+
+- [use brew to setup Python](https://docs.brew.sh/Homebrew-and-Python)
+- `pip3 install -r docs/requirements.txt`
```

Comment:
I think the installation of requirements.txt is not especially for macOS users. 

`pip install -r docs/requirements.txt` can also work on Windows and Linux. note: sometimes it is `pip` rather than `pip3`.